src/NHibernate/Type/TimeAsTimeSpanType.cs

```diff
@@ -43,10 +43,11 @@ public override object Get(DbDataReader rs, int index, ISessionImplementor sessi
 			try
 			{
 				var value = rs[index];
-				if(value is TimeSpan time) //For those dialects where DbType.Time means TimeSpan.
+				if (value is TimeSpan time) //For those dialects where DbType.Time means TimeSpan.
 					return time;
-                
-				return ((DateTime)value).TimeOfDay;
+
+				var dbValue = Convert.ToDateTime(value);
```

Comment:
The direct cast was not failing thanks to having previously used the "internal use only" `DATETIME` type, causing the data provider to do the convert itself. But this was also wrecking casting, and as documented in their [chm](https://system.data.sqlite.org/index.html/doc/trunk/Doc/SQLite.NET.chm?mimetype=application/x-chm), this type is meant for being used outside of the provider itself.

So we need know to do the convert, as are already doing other NHibernate types dealing with `DateTime`, see [`TimeType`](https://github.com/nhibernate/nhibernate-core/blob/25bc518461b71809d5701851468be0b9a550f832/src/NHibernate/Type/TimeType.cs#L60) by example.